src/main/scala/chisel3/util/random/LFSR.scala

```diff
@@ -47,19 +47,24 @@ trait LFSR extends PRNG {
     */
   def reduction: LFSRReduce
 
-  seed match {
-    case Some(s) =>
-      reduction match {
+  override protected def resetValue: Vec[Bool] = seed match {
+    case Some(s) => reduction match {
         case XOR  => require(s != 0, "Seed cannot be zero")
         case XNOR => require(s != BigInt(2).pow(width) - 1, "Seed cannot be all ones (max value)")
       }
-    case None =>
+      super.resetValue
+
+    case None => {
+      val res = WireDefault(Vec(width, Bool()), DontCare)
+
       reduction match {
-        case XOR  => when (reset.toBool) { state(0) := 1.U }
-        case XNOR => when (reset.toBool) { state(0) := 0.U }
+        case XOR  => res(0) := true.B
+        case XNOR => res(0) := false.B
```

Comment:
Use of `true.B`/`false.B` is cleaner than my original `1.U`/`0.U`. Thanks.

Reply:
Haha, I wasn't intentionally changing that; it's a result of deleting the old code and rewriting it from scratch.